pennylane/ftqc/ftqc_device.py

```diff
@@ -0,0 +1,173 @@
+        self.capabilities = DeviceCapabilities.from_toml_file(self.config_filepath)
+
+    def preprocess_transforms(self, execution_config=DefaultExecutionConfig):
+
+        program = qml.transforms.core.TransformProgram()
+
+        # validate the initial circuit is one we can process + update all observables to have wires
+        program.add_transform(no_analytic)
+        program.add_transform(validate_device_wires, wires=self.wires, name=self.name)
+        program.add_transform(
+            validate_observables,
+            lambda obs: self.capabilities.supports_observable(obs.name),
+            name=self.name,
+        )
+
+        # convert to mbqc formalism
+        program.add_transform(split_non_commuting)
+        program.add_transform(measurements_from_samples)
+
+        program.add_transform(convert_to_mbqc_gateset)
+        program.add_transform(combine_global_phases)
+        program.add_transform(convert_to_mbqc_formalism)
+
+        # validate that conversion didn't use too many wires
+        program.add_transform(
+            validate_device_wires,
+            wires=self.backend.wires,
+            name=f"{self.name}.{self.backend.name}",
+        )
+
+        # set up for backend execution (including MCM handling)
+        if self.backend.diagonalize_mcms:
+            program.add_transform(diagonalize_mcms)
+        # backend preprocess will include mcm execution method if relevant
+        backend_program, _ = self.backend.device.preprocess(execution_config)
+
+        # we skip gradient preprocess transforms, not worrying about derivatives for this prototype
+
+        return program + backend_program
+
+    def setup_execution_config(
+        self, config: ExecutionConfig | None = None, circuit: QuantumScript | None = None
+    ) -> ExecutionConfig:
+        """Sets up an ``ExecutionConfig`` that configures the execution behaviour.
+
+        In this case, the only modification compared to the standard `ExecutionConfig` is
+        that it gets the MCM method for the backend toml file (either "device" or "one-shot"),
+        so that it can be included when building the transform program for the backend.
+        uses the PennyLane device API.
+
+        Args:
+            config (ExecutionConfig): The initial ExecutionConfig object that describes the
+                parameters needed to configure the execution behaviour.
+            circuit (QuantumScript): The quantum circuit to customize the execution config for.
+
+        Returns:
+            ExecutionConfig: The updated ExecutionConfig object
+
+        """
+
+        if config is None:
+            config = ExecutionConfig()
+
+        # get mcm method - "device" if its an option, otherwise "one-shot"
+        default_mcm_method = _default_mcm_method(self.backend.capabilities, shots_present=True)
+        new_mcm_config = replace(config.mcm_config, mcm_method=default_mcm_method)
+        config = replace(config, mcm_config=new_mcm_config)
```

Comment:
Since replace creates a new object (likely not an issue for now), is there a way we could defer the creation of the dataclass to have it build with the correct values? This should be possible for the None case, no?